### PR TITLE
142552075 lucy kyle see assignees

### PR DESCRIPTION
**Pull request by kheppenstall**
Add the routing and functionality for links.

-Routes and router are currently defined in App.js (any thoughts on refactoring this?)
-Links in the sidebar are functional (for the ones we have pages for)
-Clicking on a names takes you to `/johari/:id` where that id is that users id

### CODE REVIEW

**Comment by kheppenstall** [comment]
@lucyconklin @akintner I made some progress on this but still have the problem of not accessing the name. The way I understand the problem / conventions is that this url should be accessible regardless of what pages we were on previously. So, visiting straight to `/johari/4` should bring me to my johari page for the user with an ID of 4 (if I have that assignment). However, then we will have now way of knowing user 4's name without pinging the API. So I'm back to thinking we'll have to hit the API to get the user's name. As it stands, the id of the user is saved as a prop of `evaluateeID` on the johari component. I had to make some changes to how we defined our routes in order to figure out how to pass the property through. They are defined more explicitly now instead of with the map enumerable.

File: src/assignee-list/AssigneeList.js [hunk 1]
```diff
@@ -9,11 +9,14 @@ class AssigneeList extends Component {
   }
 
   componentDidMount() {
-    this.setState({ assignees: ["Self", "Annie", "Kyle", "Drew" ]});
+    var that = this;
```
Comment (lucyconklin) [inline src/assignee-list/AssigneeList.js]: We can get rid of the this/that line actually. I just updated and will push with the next commit. Thanks ES6!

Comment (kheppenstall) [reply src/assignee-list/AssigneeList.js]: @lucyconklin That is wonderful!

**Comment by lucyconklin** [comment]
1. @kheppenstall  I totally agree that we should pull the router out somewhere.
@case-eee what is the convention here? It makes the most sense to me that we make some kind of router like './src/router.js' that handles the routing and is called in App.js, and is not a component like the others are?

2. Let's proceed as you say above by hitting the API for username. I still stubbornly think that we can do this with just one API call, but right now we should just get this functionality up. I will let it marinate and see if I can come up with a solution. Plus, the best solution might be something cool like caching, or taking advantage of a React feature I don't know about.

**Comment by kheppenstall** [comment]
@lucyconklin Cool that sounds great. For 2., Another option might be to have the endpoints of our API match the pages - so `johari/2` would respond with the adjectives and the evaluatee's name.

**Comment by case-eee** [comment]
@lucyconklin i'm not sure what the best practices are for your router. what you've suggested definitely makes sense to me. i'm happy to reach out to Brenna/Jhun to ask their thoughts (since they teach React in the FE program). what do you think?

**Pull request merged**